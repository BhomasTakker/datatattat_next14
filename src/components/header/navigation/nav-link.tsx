import Link from "next/link";

import styles from "./nav-link.module.scss";

export type NavLinkData = {
	route: string;
	label: string;
	key?: string;
};

type NavLinkProps = { isEdit: boolean } & NavLinkData;

export const NavLink = ({ route, label, isEdit }: NavLinkProps) => {
	const routeLink = isEdit ? `/edit?route=${route}` : route;
	// if edit...
	return (
		<Link className={`${styles.root}`} href={routeLink}>
			<p>{label}</p>
		</Link>
	);
};
