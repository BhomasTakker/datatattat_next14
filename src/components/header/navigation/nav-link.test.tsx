import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { NavLink } from "./nav-link";

describe("Nav Link", () => {
	it("renders a link", () => {
		render(<NavLink route="/home" label="Home" />);
		const link = screen.getByRole("link", {
			name: /home/i,
		});
		expect(link).toBeInTheDocument();
	});
	it("renders a link with the provided href", () => {
		render(<NavLink route="/home" label="Home" />);
		const link = screen.getByRole("link", {
			name: /home/i,
		});
		expect(link).toHaveProperty("href", "http://localhost/home");
	});
	it("renders nav link unchanged", () => {
		const { container } = render(<NavLink route="/home" label="Home" />);
		expect(container).toMatchSnapshot();
	});
});
