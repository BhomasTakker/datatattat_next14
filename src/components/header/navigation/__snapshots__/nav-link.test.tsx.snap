// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Link renders nav link unchanged 1`] = `
<div>
  <div
    class="root"
  >
    <a
      class="link"
      href="/home"
    >
      <p>
        Home
      </p>
    </a>
  </div>
</div>
`;
